daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/parsers/PState.scala

```diff
@@ -288,7 +285,8 @@ final class PState private (
   }
 
   def currentLocation: DataLocation = {
-    val isAtEnd = !dataInputStream.isDefinedForLength(1)
+    // val isAtEnd = !dataInputStream.isDefinedForLength(1)
+    val isAtEnd = dataInputStream.isAtEnd()
```

Comment:
Arguably, isAtEnd should not be captured into this object at this point, and frozen. 

Because it could be false, yet a subsequent operation on the InputSourceDataInputStream to look and see if there is more data could get back End-of-Data, which would change the answer, that yes, in fact, the parse did end at end of data.

Reply:
Agreed, but need to consider backwards compatibility.

Reply:
In the latest, I have deprecated isAtEnd entirely. The definition that remains is not compatible backward because it does not probe the data to see if we have more, or are at EOF.  

I'm wondering if we should put back in that blocking behavior, but now that this is deprecated, that would allow people who still have this code to work the same, while prompting them to rewrite it via the deprecation warning.

Reply:
Latest commit to this PR puts back isAtEnd (though still deprecated) with the original blocking behavior. 

Daffodil itself doesn't use this. It's purely to retain compatibility.